db/models/users.js

```
@@ -18,4 +18,50 @@ module.exports = {
 			.update({ verified: true })
 			.returning('userID');
 	},
+	getUserPayout(id) {
+		return db('Users as u')
+			.join('Offers as o', 'u.id', 'o.userID')
+			.where('u.id', id)
+			.andWhere('o.active', true)
+			.sum('o.userOfferInCents');
+	},
+	async addLicense(license) {
+		let userLicense = await db('UserLicenses').where({ userID: license.userID }).first();
+		if (!userLicense) return db('UserLicenses').insert(license);
+		else return db('UserLicenses').where({ userID: license.userID }).update(license);
+	},
+	async getActiveOffers(id) {
+		let orders = await db('Offers as o')
+			.join('Users as u', 'u.id', 'o.userID')
+			.join('Merchants as m', 'm.id', 'o.merchantID')
+			.join('GiftCards as gc', 'gc.id', 'o.giftCardID')
+			.where('u.id', id)
+			.andWhere('o.active', true)
+			.select(
+				'o.id',
+				'o.userID',
+				'o.merchantID',
+				'gc.cardNumber',
+				'm.name as merchantName',
+				'o.userOfferInCents',
+				'o.createdAt',
+			);
+
+		if (!orders.length) {
+			let error = new Error('No active orders found for user');
+			error.status = 404;
+			throw error;
+		}
+		return orders;
+	},
+	async getUserLicense(userID) {
+		let license = await db('UserLicenses').where({ userID }).first();
+
+		if (!license) {
+			let error = new Error('No license information found for user');
+			error.status = 401;
+			throw error;
```

Don't think we want to throw, but I think I agree that we should return an error code

---

what's the difference?

---

throwing an exception is computationally expensive and prevents any further processing from occurring by crashing the app

---

i guess i'm confused by this, it doesn't 'crash' it, it sends it to the catch block which then makes it return the error i'm defining here. i just read up on node error handling best practices and throwing seems to be the way its done in these examples 🤔

---

Ok I think you're right, I was thinking about swift/objc/java exceptions